src/jit/lowerxarch.cpp

```diff
@@ -2191,8 +2191,26 @@ void Lowering::TreeNodeInfoInitModDiv(GenTree* tree)
         info->setDstCandidates(l, RBM_RAX);
     }
 
-    // If possible would like to have op1 in RAX to avoid a register move
-    op1->gtLsraInfo.setSrcCandidates(l, RBM_RAX);
+#ifdef _TARGET_X86_
+    if (op1->OperGet() == GT_LONG)
+    {
+        GenTree* loVal = op1->gtGetOp1();
+        GenTree* hiVal = op1->gtGetOp2();
+        
+        // Src count is actually 3, so increment.
+        info->srcCount++;
+
+        // Op1's low part should be in RAX, and its high part in RDX.
+
+        loVal->gtLsraInfo.setSrcCandidates(l, RBM_RAX);
+        hiVal->gtLsraInfo.setSrcCandidates(l, RBM_RDX);
+    }
```

Comment:
You should use EAX/EDX for x86 instead of RAX/RDX (even though the #defines are the same)